tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/planner_utils.h

```diff
@@ -101,6 +104,51 @@ bool isValidState(const tesseract_kinematics::ForwardKinematics::ConstPtr& robot
   return !(robot_config != RobotConfig::FUT && robot_config != RobotConfig::NUT);
 }
 
+/**
+ * @brief Get the profile string taking into account defaults and profile remapping
+ * @param profile The requested profile name in the instructions
+ * @param planner_name Used to look up if there are remappings available
+ * @param profile_remapping Remapping used to remap a profile name based on the planner name
+ * @param default_profile Default = DEFAULT. This is set if profile.empty()
+ * @return The profile string taking into account defaults and profile remapping
+ */
+inline std::string getProfileString(const std::string& profile,
+                                    const std::string& planner_name,
+                                    const PlannerProfileRemapping& profile_remapping,
+                                    std::string default_profile = "DEFAULT")
+{
+  std::string results = profile;
+  if (profile.empty())
+    results = std::move(default_profile);
+
+  // Check for remapping of profile
+  auto remap = profile_remapping.find(planner_name);
+  if (remap != profile_remapping.end())
+  {
+    auto p = remap->second.find(profile);
+    if (p != remap->second.end())
+      results = p->second;
+  }
+  return results;
+}
+
+template <typename ProfileType>
+std::shared_ptr<ProfileType>
+getProfile(const std::string& profile, const std::unordered_map<std::string, std::shared_ptr<ProfileType>>& profile_map)
+{
+  std::shared_ptr<ProfileType> results;
```

Comment:
recommend changing this to `auto results = std::make_shared<ProfileType>();` this way it will return a profile with default setting if one is not found.

Reply:
Ah. I forgot to set it to DEFAULT if it wasn't found. That should fix it.

Reply:
Also, I can't do make_shared because ProfileType is an abstract class.